python/cudf/dataframe/multiindex.py

```diff
@@ -200,9 +246,16 @@ def _get_column_major(self, df, row_tuple):
         return result
 
     def __len__(self):
+        if self._codes is None:
+            if self._df is not None:
+                self._compute_levels_and_codes()
+            else:
+                return 0
         return len(self.codes[self.codes.columns[0]])
 
     def __eq__(self, other):
+        if self._codes is None:
+            self._compute_levels_and_codes()
```

Comment:
Here too

Reply:
This one is tricky - implicitly codes get computed during __eq__, and __eq__ is calculated when indices are set on columns. I'm not sure what the fix is, take a look?

Reply:
Instead of comparing the `codes` and `levels` of a MultiIndex, we can compare the actual columns to generate a few boolean columns, and then do some logical ops to get a scalar boolean of whether they're equal or not.